src/Tribe/JSON_LD/Order.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+// Don't load directly
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+/**
+ * A JSON-LD class to hook and change how Orders work on Events
+ * @todo rework this class to make it standalone from The Events Calendar
+ */
+class Tribe__Tickets__JSON_LD__Order {
+
+	/**
+	 * Get (and instantiate, if necessary) the instance of the class
+	 *
+	 * @static
+	 * @return self
+	 *
+	 */
+	public static function instance() {
+		static $instance;
+
+		if ( ! $instance instanceof self ) {
+			$instance = new self;
+		}
+
+		return $instance;
+	}
+
+	/**
+	 * Value indicating low stock availability for a specific ticket.
+	 *
+	 * This can be overridden with the tribe_tickets_json_ld_low_inventory_level filter.
+	 *
+	 * @var int
+	 */
+	protected $low_stock = 5;
+
+	/**
+	 * Setup Google Event Data for tickets.
+	 */
+	public static function hook() {
```

Comment:
> There are no hooks on the constructor, just setting values.

Yep, I'm blind.

> I always like to not do anything on instance() because it prevents re-attaching the functions to the filters/actions.

I guess what I mean is, it's conventional to get the instance via the `instance()` method: but here there are three different public methods that can all be used to obtain an `Tribe__Tickets__JSON_LD__Order` object (with different results in each case).

Wouldn't it be cleaner/tighten things up to use an arrangement more like this?

```
pubic static instance();   // Calls the constructor *and* hook()
protected static hook();   // Cannot be called directly
protected __constructor(); // Cannot be called directly
```